server/src/main/java/io/spine/server/reflect/FailureSubscriberMethod.java

```diff
@@ -212,6 +223,95 @@ private static Factory getInstance() {
     }
 
     /**
+     * A {@link Function} wrapping the given {@link Method} into a corresponding
+     * {@code FailureSubscriberMethod}.
+     */
+    private enum MethodWrapper implements Function<Method, FailureSubscriberMethod> {
+
+        /**
+         * Wraps the given {@link Method} with an instance of {@link ShortFailureSubscriberMethod}.
+         */
+        SHORT {
+            @Override
+            FailureSubscriberMethod wrap(Method method) {
+                return new ShortFailureSubscriberMethod(method);
+            }
+        },
+
+        /**
+         * Wraps the given {@link Method} with an instance of
+         * {@link CommandContextAwareFailureSubscriberMethod}.
+         */
+        COMMAND_CONTEXT_AWARE {
+            @Override
+            FailureSubscriberMethod wrap(Method method) {
+                return new CommandContextAwareFailureSubscriberMethod(method);
+            }
+        },
+
+        /**
+         * Wraps the given {@link Method} with an instance of
+         * {@link CommandMessageAwareFailureSubscriberMethod}.
+         */
+        COMMAND_MESSAGE_AWARE {
+            @Override
+            FailureSubscriberMethod wrap(Method method) {
+                return new CommandMessageAwareFailureSubscriberMethod(method);
+            }
+        },
+
+        /**
+         * Wraps the given {@link Method} with an instance of
+         * {@link CommandAwareFailureSubscriberMethod}.
+         */
+        COMMAND_AWARE {
+            @Override
+            FailureSubscriberMethod wrap(Method method) {
+                return new CommandAwareFailureSubscriberMethod(method);
+            }
+        };
+
+        /**
+         * Retrieves the wrapper implementation for the given set of parameters of a handler method.
+         *
+         * @param paramTypes the types of the parameters of the handler method
+         * @return the corresponding instance of {@code MethodWrapper}
+         */
+        private static MethodWrapper forParamSet(Class[] paramTypes) {
+            checkNotNull(paramTypes);
+            final int paramCount = paramTypes.length;
+            final MethodWrapper methodWrapper;
+            switch (paramCount) {
+                case 1:
+                    methodWrapper = SHORT;
+                    break;
+                case 2:
+                    final Class<?> secondParamType = paramTypes[1];
+                    methodWrapper = secondParamType == CommandContext.class
+                                    ? COMMAND_CONTEXT_AWARE
+                                    : COMMAND_MESSAGE_AWARE;
+                    break;
+                case 3:
+                    methodWrapper = COMMAND_AWARE;
+                    break;
+                default:
+                    throw new IllegalArgumentException(
```

Comment:
By the invariant of the process of scanning the Subscriber methods by the `Bus`, this branch is unreachable.
I see no possibility to test it except changing the visibility of the `FailureSubscriberMethod` nested types.